pkg/sr/serde.go

```diff
@@ -172,7 +183,16 @@ func (s *Serde) Register(id int, v any, opts ...SerdeOpt) {
 	// tree to find the end node we are initializing.
```

Comment:
Can you add a `defer` before s.ids.Store(m) on line L168? I think the code is racy (before this PR)

Reply:
It wasn't as simple, because we change `m` below. If we are registering a schema with indices, `m` ends up pointing to the last map of subindices.

I added a variable `dupIDs` to match `dupTypes` and store both of them in the same deferred call.